### PR TITLE
fix: fix snake case env var

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

These two particular GH env vars are snake case.......

### CODE REVIEW

File: .github/workflows/scripts/e2e-verify.common.sh [hunk 1]
```diff
@@ -85,8 +85,8 @@ e2e_verify_common_buildDefinition_v1() {
     e2e_verify_predicate_v1_buildDefinition_systemParameters "$1" "GITHUB_REPOSITORY_OWNER_ID" "$GITHUB_REPOSITORY_OWNER_ID"
     e2e_verify_predicate_v1_buildDefinition_systemParameters "$1" "GITHUB_WORKFLOW_REF" "$GITHUB_WORKFLOW_REF"
     e2e_verify_predicate_v1_buildDefinition_systemParameters "$1" "GITHUB_WORKFLOW_SHA" "$GITHUB_WORKFLOW_SHA"
-    e2e_verify_predicate_v1_buildDefinition_systemParameters "$1" "IMAGE_OS" "$IMAGE_OS"
-    e2e_verify_predicate_v1_buildDefinition_systemParameters "$1" "IMAGE_VERSION" "$IMAGE_VERSION"
+    e2e_verify_predicate_v1_buildDefinition_systemParameters "$1" "IMAGE_OS" "$ImageOs"
```
Comment (ianlewis) [inline .github/workflows/scripts/e2e-verify.common.sh:88]: This seems to be `ImageOS`?
```suggestion
    e2e_verify_predicate_v1_buildDefinition_systemParameters "$1" "IMAGE_OS" "$ImageOS"
```

**Review by ianlewis — APPROVED** [review]
You can shellcheck comments also to ignore the errors/warnings about the variables being set.

*shellcheck assumes non-capitalized variables should be local.

**Comment by laurentsimon** [comment]
Interesting. Shall we respect the original variable case-sensitivity? Wha's GitHub team's view?
/cc @kommendorkapten

**Comment by asraa** [comment]
> Interesting. Shall we respect the original variable case-sensitivity? What's GitHub team's view?

Seems like they also include the var names in the provenance as `IMAGE_OS` and `IMAGE_VERSION`. But yeah, I don't know why those in particular were snake cased

**Comment by kommendorkapten** [comment]
As @asraa said, npm CLI extracts them like this:

```
          IMAGE_OS: env.ImageOS,
          IMAGE_VERSION: env.ImageVersion,
```

I think SLSA generator should do the same, I have also asked if we can get environment variables with a consistent name too so we don't need to transform them. 

Only one concern, would we pose a risk from a reproducible build scenario by transforming them? I think the risk is negligible, as JavaScript/npm is virtually not possible to build reproducible. If it's more for capturing the provenance, I would favour consistency in the variable names.

**Comment by ianlewis** [comment]
> Only one concern, would we pose a risk from a reproducible build scenario by transforming them? I think the risk is negligible, as JavaScript/npm is virtually not possible to build reproducible. If it's more for capturing the provenance, I would favour consistency in the variable names.

`systemParameters` don't have to be 1:1 environment variable names so I think we're fine here as long as there is an ability to understand what parameters like `IMAGE_OS` and `IMAGE_VERSION` meant.

**Pull request merged**